### ISSUE TITLE
5.5.0-dev.20240524 regression on "The inferred type of 'X' cannot be named without a reference"

### ISSUE THREAD

**Issue by jgoux** [microsoft/TypeScript#58710]
### 🔎 Search Terms

cannot be named without a reference

### 🕗 Version & Regression Information

- This changed between versions 5.5.0-dev.20240523 and 5.5.0-dev.20240524

I also tried running the latest version `5.5.0-dev.20240530` and the error is still reported.

### 💻 Code

This file https://github.com/jgoux/acme/blob/main/apps/api/src/config/trpc.ts is reported with an error that didn't happen before `typescript@5.5.0-dev.20240524`.

You can try running the type-checking command in the root of the monorepo: `pnpm type-check`.

### 🙁 Actual behavior

This error happens:

```
TS2742: The inferred type of 't' cannot be named without a reference to '../../node_modules/@trpc/server/dist/core/internals/config.js'. This is likely not portable. A type annotation is necessary.

3 export const t = initTRPC.create();
```

### 🙂 Expected behavior

The error shouldn't be reported.

**Comment by Andarist** [comment]
It would be great if u could post what dts was generated there in 5.4

**Comment by jgoux** [comment]
> * 5.5.0-dev.20240523

Here are the produced dts files for both `5.4.5` and `5.5.0-dev.20240523` (last working version): https://gist.github.com/jgoux/aec06283fb607c1c850ef1c9025af0c4
